products/views.py

```diff
@@ -1,3 +1,192 @@
+            # If page is not an integer, deliver first page
+            search_results = paginator.page(1)
+        except EmptyPage:
+            # If page is out of range (e.g. 9999), deliver last page of results
+            search_results = paginator.get_page(paginator.num_pages)
+
+    return render(request, "search.html", {"search_results": search_results})
+
 
 # Create your views here.
+@login_required
+def add_to_cart(request, book_title):
+    book = Book.objects.get(title=book_title)
+    try:
+        cart = Cart.objects.get(created_by=request.user)
+        try:
+            cartItem = CartItems.objects.get(cart=cart, book=book)
+            cartItem.quantity += 1
+            cartItem.save()
+            return HttpResponse("You have successfully")
+        except CartItems.DoesNotExist:
+            cartItem = CartItems.objects.create(cart=cart, book=book, quantity=1)
+            cartItem.save()
+            return HttpResponse("Book added to your cart")
+    except Cart.DoesNotExist:
+        timeNow = datetime.now()
+        cart = Cart.objects.create(
+            created_by=request.user,
+            created_at=timeNow
+        )
+        cartItem = CartItems.objects.create(
+            cart=cart,
+            book=book,
+            quantity=1
+        )
+        cartItem.save()
+        return HttpResponse("Your cart have been added")
+
+
+def remove_from_cart(request, slug):
+    book = get_object_or_404(Book, slug=slug)
+    cart = Cart.objects.get(created_by=request.user)
+    try:
+        cartItem = CartItems.objects.get(cart=cart, book=book)
+        if cartItem.quantity == 0:
+            cartItem.delete()
+            messages.info(request, "Item deleted successfully")
+        else:
+            cartItem.quantity -= 1
+            cartItem.save()
+            messages.info(request, "Item removed successfully")
+    except CartItems.DoesNotExist:
+        messages.info(request, 'You do not have the product in cart')
+
+
+
+
+class OrderSummaryView(View):
+    def get(self, *args, **kwargs):
+        cart = get_object_or_404(Cart, created_by=self.request.user)
+        cartItems = CartItems.objects.filter(cart=cart)
+        totalPrice = 0
+        for item in cartItems:
+            book = Book.objects.get(title=item.book.title)
+            totalPrice += item.quantity * book.price
+        context = {
+            'cart_items': cartItems,
+            'orders': cart,
+            'total_price': totalPrice
+        }
+        return render(self.request, 'order_summary.html', context)
+
+
+class PaymentView(View):
+    def get(self, *args, **kwargs):
+        order = Order.objects.filter(user=self.request.user).first()
+        if order.address:
+            context = {
+                'order': order
+            }
+            return render(self.request, 'payment.html')
+        else:
+            messages.info('You do not have an billing address yet')
+            return redirect('checkout')
```

Comment:
Right now this is unused (and is also broken), so either delete or fix it if you use it later.

Reply:
I intend to use it to do Stripe and Paypal